File: Problems/011-container-with-most-water/container-with-most-water.go
package problem011

func maxArea(height []int) int {
	area, temp := 0, 0
	i, j := 0, len(height)-1

	for i < j {
		if height[i] < height[j] {
			temp = height[i] * (j - i)
			i++
		} else {
			temp = height[j] * (j - i)
			j--
		}

		if temp > area {
			area = temp
		}
	}

	return area
}

File: Problems/011-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, len(height)-1
        area = 0

        while i < j:
            temp = min(height[i], height[j]) * (j-i)
            if temp > area:
                area = temp
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        
        return area
            